arangod/Transaction/Methods.cpp

```diff
@@ -3139,17 +3138,19 @@ Future<Result> Methods::replicateOperations(
     auto& rtc = static_cast<ReplicatedRocksDBTransactionCollection&>(
         transactionCollection);
     auto leaderState = rtc.leaderState();
-    try {
-      leaderState->replicateOperation(
-          replicationData.sharedSlice(),
-          replication2::replicated_state::document::fromDocumentOperation(
-              operation),
-          state()->id(), rtc.collectionName(),
-          replication2::replicated_state::document::ReplicationOptions{});
-    } catch (basics::Exception const& e) {
-      return Result{e.code(), e.what()};
-    } catch (std::exception const& e) {
-      return Result{TRI_ERROR_INTERNAL, e.what()};
+    auto replicatedOp = replication2::replicated_state::document::
+        ReplicatedOperation::buildDocumentOperation(
+            operation, state()->id(), rtc.collectionName(),
+            replicationData.sharedSlice());
+    // Should finish immediately
+    auto replicationRes =
+        leaderState
+            ->replicateOperation(
+                std::move(replicatedOp),
+                replication2::replicated_state::document::ReplicationOptions{})
+            .get();
+    if (replicationRes.fail()) {
+      return replicationRes.result();
```

Comment:
Why should this finish immediately? Shouldn't this be async?

Reply:
Same as above